src/ray/core_worker/core_worker.cc

```diff
@@ -379,7 +379,8 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
       num_executed_tasks_(0),
       task_execution_service_work_(task_execution_service_),
       resource_ids_(new ResourceMappingType()),
-      grpc_service_(io_service_, *this) {
+      grpc_service_(io_service_, *this),
+      exiting_(false) {
```

Comment:
This doesn't seem to be used?

Reply:
(it never became true)

Reply:
ah it's an existing variable with race condition.